Godeps/_workspace/src/github.com/docker/distribution/registry/storage/layerstore.go

```diff
@@ -52,6 +52,18 @@ func (ls *layerStore) Fetch(dgst digest.Digest) (distribution.Layer, error) {
 	}, nil
 }
 
+func (ls *layerStore) Delete(dgst digest.Digest) error {
+	lp, err := ls.linkPath(dgst)
+	if err != nil {
+		return err
+	}
+
+	// strip off /link at the end
+	lp = lp[:len(lp)-5]
```

Comment:
Do you not want to check the suffix before doing this?  I recently got in trouble making an assumption and doing the same thing.